### PR TITLE
MM-47561 : fix icons alignment in global threads

### PR DESCRIPTION
**Pull request by AbhinavVihan**
#### Summary
fix icons alignment in global threads

#### Ticket Link
Fixes: https://github.com/mattermost/mattermost-server/issues/21402
JIRA: https://mattermost.atlassian.net/browse/MM-47561

#### Related Pull Requests
None

#### Release Note
None

```release-note
NONE
```

### CODE REVIEW

**Review by michelengelen — APPROVED** [review]
lgtm

File: components/threading/global_threads/thread_item/__snapshots__/thread_item.test.tsx.snap [hunk 1]
```diff
@@ -54,7 +54,7 @@ exports[`components/threading/global_threads/thread_item should report total num
           className="Button___icon"
         >
           <i
-            className="Icon icon-dots-vertical"
+            className="icon icon-dots-vertical"
```
Comment (AshishDhama) [inline components/threading/global_threads/thread_item/__snapshots__/thread_item.test.tsx.snap:57]: @AbhinavVihan Check if we can use compass icon here

**Review by AshishDhama — CHANGES_REQUESTED** [review]

File: package-lock.json [hunk 2]
```diff
@@ -5,6 +5,7 @@
   "requires": true,
   "packages": {
     "": {
+      "name": "mattermost-webapp",
```
Comment (M-ZubairAhmed) [inline package-lock.json:8]: please revert this change

**Comment by AbhinavVihan** [comment]
@M-ZubairAhmed, the requested changes are done.

File: components/threading/common/button/button.tsx [hunk 3]
```diff
@@ -35,7 +37,7 @@ function Button({
                     {prepend}
                 </span>
             )}
-            <span className='Button_label'>
+            <span className={marginExtra ? 'Button_label_margin_extra' : 'Button_label'}>
```
Comment (M-ZubairAhmed) [inline components/threading/common/button/button.tsx:40]: after the above change we can just use `className('Button_label, {'margin_extra': marginExtra}`

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]

File: components/threading/global_threads/thread_list/thread_list.tsx [hunk 4]
```diff
@@ -227,9 +229,10 @@ const ThreadList = ({
                                 disabled={!someUnread}
                                 className={'Button___large Button___icon'}
                                 onClick={handleOpenMarkAllAsReadModal}
+                                marginExtra={true}
                             >
-                                <span className='Icon'>
-                                    <i className='icon-playlist-check'/>
+                                <span className='icon'>
```
Comment (M-ZubairAhmed) [inline components/threading/global_threads/thread_list/thread_list.tsx:234]: why the classname change here to `icon`?

Comment (AbhinavVihan) [reply components/threading/global_threads/thread_list/thread_list.tsx:234]: cause with ### Icon its adding some extra margin by itself, plus we have used ### icon everywhere else

Comment (AbhinavVihan) [reply components/threading/global_threads/thread_list/thread_list.tsx:234]: > nunnu

**Review by M-ZubairAhmed — APPROVED** [review]

**Review by matthewbirtch — APPROVED** [review]
Looks good to me! Thanks for fixing this @AbhinavVihan

File: components/threading/common/button/button.tsx [hunk 5]
```diff
@@ -12,6 +12,7 @@ type Props = {
     isActive?: boolean;
     hasDot?: boolean;
     allowTextOverflow?: boolean;
+    marginExtra?: boolean;
```
Comment (AshishDhama) [inline components/threading/common/button/button.tsx:15]: @AbhinavVihan Better name would marginTop? as marginExtra is misleading.

**Review by AshishDhama — APPROVED** [review]

**Review by jgilliam17 — APPROVED** [review]
Thanks @AbhinavVihan 
Tested, looks good to merge.
Verified icon alignment in global threads.
[E2E report ](https://mattermost-cypress-report.s3.amazonaws.com/275024-2abe4c7-onprem-ent-webapp-pr-11606-mm-ee-test:2abe4c7/mochawesome.html)- no PR related failures.
